lib/inspec/profile.rb

```diff
@@ -162,10 +168,11 @@ def params
     end
 
     def collect_tests(include_list = @controls)
-      if !@tests_collected
+      unless @tests_collected
         locked_dependencies.each(&:collect_tests)
 
         tests.each do |path, content|
+          next unless supports_platform?
```

Comment:
I think you could put this before the tests array like:
```
return unless supports_platform?

tests.each do |path, content|
...
```
This will keep it from being called for every single test

Reply:
Oh right. Thank you =)